tests/utils/test_logging.py

```diff
@@ -126,14 +126,8 @@ def test_advisory_warnings(self):
 
 
 def test_set_progress_bar_enabled():
-    TINY_MODEL = "hf-internal-testing/tiny-random-distilbert"
-    with patch("tqdm.auto.tqdm") as mock_tqdm:
-        disable_progress_bar()
-        _ = AutoConfig.from_pretrained(TINY_MODEL, force_download=True)
-        mock_tqdm.assert_not_called()
+    disable_progress_bar()
+    assert are_progress_bars_disabled()
 
-        mock_tqdm.reset_mock()
-
-        enable_progress_bar()
-        _ = AutoConfig.from_pretrained(TINY_MODEL, force_download=True)
-        mock_tqdm.assert_called()
+    enable_progress_bar()
+    assert not are_progress_bars_disabled()
```

Comment:
I did not manage to adapt this test with mock since:
- the tqdm from `huggingface_hub.file_download` is always called
- the calls to tqdm.auto.tqdm are not detected anymore
If someone has a genius idea, happy to hear!

Reply:
In `huggingface_hub` I test that by using the `capsys` fixture from pytest that catch the output while running the tests. Then you can check if the output stayed quiet or if you have progress bars in it.

You can have a look [here](https://github.com/huggingface/huggingface_hub/blob/main/tests/test_utils_tqdm.py#L17) for the setup and [here](https://github.com/huggingface/huggingface_hub/blob/main/tests/test_utils_tqdm.py#L107) for a test where tqdm is enabled.

However from `transformers`'s point of view, this is a job done by an external library (`huggingface_hub`) so testing it here would be a duplicate of what is already tested. It's would not be wrong to test it but the current test you wrote is good enough I think :+1:

Reply:
And in general, are `transformers`'s `_tqdm_active` and `_tqdm_cls` used anywhere now ?

Reply:
No they're not used anywhere now (we do use tqdm in other places, so should switch to the one in logging, but probably for another PR).